mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/loader/validator/spi/ExternalValidatorFinder.java

```diff
@@ -0,0 +1,39 @@
+package org.mule.runtime.extension.internal.loader.validator.spi;
+
+import static java.util.Collections.emptyList;
+import static org.mule.runtime.extension.internal.loader.ExtensionModelFactory.COMPILATION_MODE_FLAG;
+
+import org.mule.runtime.extension.api.loader.ExtensionLoadingContext;
+import org.mule.runtime.extension.api.loader.ExtensionModelValidator;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.ServiceLoader;
+
+/**
+ * Finder of {@link ExtensionModelValidator} that are contributed using SPI.
+ *
+ * @since 1.3.0
+ */
+public class ExternalValidatorFinder {
+
+    /**
+     * Looks and find the available external {@link ExtensionModelValidator}
+     *
+     * @param extensionLoadingContext Loading context
+     * @return All the {@link ExtensionModelValidator} that are contributed using SPI in the current class loader.
+     */
+    public static List<ExtensionModelValidator> find(ExtensionLoadingContext extensionLoadingContext) {
+       return extensionLoadingContext.getParameter(COMPILATION_MODE_FLAG).map(flag -> {
+            if (flag instanceof Boolean && (Boolean) flag) {
+                List<ExtensionModelValidator> validators = new ArrayList<>();
+                ServiceLoader<ExtensionModelValidator> load = ServiceLoader.load(ExtensionModelValidator.class);
+                load.iterator().forEachRemaining(validators::add);
```

Comment:
this should be cached. If you have 5 apps with 3 extensions each, you'll be doing this 15 times.